prow/sidecar/run.go

```diff
@@ -77,32 +84,20 @@ func (o Options) Run(ctx context.Context) (int, error) {
 	interrupt := make(chan os.Signal)
 	signal.Notify(interrupt, os.Interrupt, syscall.SIGTERM)
 	go func() {
-		select {
-		case s := <-interrupt:
-			logrus.Errorf("Received an interrupt: %s, cancelling...", s)
-			cancel()
```

Comment:
This is the correct pattern and we should not regress it. This is what causes WaitForMarker() to know to stop waiting.

I suspect the problem you're trying to solve is to ensure we still do the upload, which right now will not happen because ctx has ben cancelled.

We should solve this by using different contexts for the upload (which we do not want to cancel) and wait (which we do).

Reply:
Why do we need to gracefully terminate waiting on markers when we get SIGTERM?

Reply:
> I suspect the problem you're trying to solve is to ensure we still do the upload, which right now will not happen because ctx has ben cancelled.

Right now we do still upload when the context has been cancelled. The problem is that we stop waiting for the marker files and just upload immediately before the markers or process logs have been written so the `build-log.txt` that gets uploaded is something like `Failed to open /logs/process-log.txt: open /logs/process-log.txt: no such file or directory`.

> We should solve this by using different contexts for the upload (which we do not want to cancel) and wait (which we do).

I don't think we want to cancel the wait. We need to keep waiting so that we can reupload once the marker files have been written to because only then do we know that the process logs have been written.

Reply:
Why doesn't the file exist? Let's fix that.

What is sending a terminate signal? Why do we want to continue waiting in this scenario (as opposed to finishing as quickly as possible)?

Reply:
The `kubelet` sends `SIGTERM` to both containers in the `Pod` -- we want to:

1. start uploading everything we've got ASAP
2. if we notice `entrypoint` honor the `SIGTERM` and publish the output of the test process, do normal upload

Both of those will race with each other and against time (the graceful deletion window from the `kubelet`)

If we FINISH uploading after the marker file, sure, we can exit

Reply:
(1) sounds good, but I disagree that we should do parallel uploads of the same data. I do not think this is the cleanest solution to the problem.